src/preprocessor.js

```diff
@@ -0,0 +1,49 @@
+import { templateResource } from "./templates";
+import { expressResource } from "./expressions";
```

Comment:
Let's rename these methods. They should be verbs as they are mutating the resource.
`templateResource` to `expandTemplates`
`expressResource` to `evaluateExpressions`
`pathResource` to `buildPath`
Thoughts?